terraform/modules/prom-ec2/paas-config/prometheus.conf.tpl

```diff
@@ -4,26 +4,67 @@ global:
 alerting:
   alertmanagers:
   - scheme: http
-    static_configs:
-      - targets: [${alertmanager_dns_names}]
+    ec2_sd_configs:
+      - region: eu-west-1
+        port: 9093
+    relabel_configs:
+      - source_labels: ['__meta_ec2_tag_Name']
+        regex: '${environment}-ecs-instance'
+        action: keep
   - scheme: https
     static_configs:
       - targets: [${external_alertmanagers}]
 rule_files:
   - "/etc/prometheus/alerts/*"
 scrape_configs:
   - job_name: prometheus
-    static_configs:
-      - targets: [${prometheus_addresses}]
+    ec2_sd_configs:
+      - region: eu-west-1
+        port: 9090
+    relabel_configs:
+      - source_labels: ['__meta_ec2_tag_Environment']
+        regex: '${environment}'
+        action: keep
+      - source_labels: ['__meta_ec2_tag_Job']
+        regex: 'prometheus'
+        action: keep
+      - source_labels: ['__meta_ec2_availability_zone']
+        target_label: availability_zone
+      - source_labels: ['__meta_ec2_instance_id']
+        replacement: '$1:9090'
+        target_label: instance
   - job_name: paas-targets
     scheme: http
     proxy_url: 'http://localhost:8080'
     file_sd_configs:
       - files: ['/etc/prometheus/targets/*.json']
         refresh_interval: 30s
   - job_name: alertmanager
-    static_configs:
-      - targets: [${alertmanager_dns_names}]
+    ec2_sd_configs:
+      - region: eu-west-1
+        port: 9093
+    relabel_configs:
+      - source_labels: ['__meta_ec2_tag_Name']
+        regex: '${environment}-ecs-instance'
+        action: keep
+      - source_labels: ['__meta_ec2_availability_zone']
+        target_label: availability_zone
+      - source_labels: ['__meta_ec2_instance_id']
+        replacement: '$1:9093'
+        target_label: instance
   - job_name: prometheus_node
-    static_configs:
-      - targets: [${prometheus_node_addresses}]
+    ec2_sd_configs:
+      - region: eu-west-1
+        port: 9100
+    relabel_configs:
+      - source_labels: ['__meta_ec2_tag_Environment']
+        regex: '${environment}'
+        action: keep
+      - source_labels: ['__meta_ec2_tag_Job']
+        regex: '.+'
```

Comment:
Is this regex intentional here? `regex` by default should apply use `.*` But if we only want to get Prometheus nodes (rather than ECS which have no Job label?) then could we more explicit and explicitly ask for instances with  job equal to 'prometheus'?

Reply:
The thinking was: match any instance which has a Job label at all (which will therefore exclude ECS).  However, Prometheus doesn't distinguish between an empty label and no label at all, so if you match on `.*` you'll match everything.

I understand about matching `prometheus` explicitly, I guess I was thinking about future-proofing but, er, YAGNI :).  Happy to just match `prometheus`.